common/src/main/java/org/apache/nemo/common/ir/IRDAGChecker.java

```diff
@@ -429,7 +434,11 @@ void addScheduleGroupCheckers() {
 
   void addEncodingCompressionCheckers() {
     final NeighborChecker additionalOutputEncoder = ((irVertex, inEdges, outEdges) -> {
-      for (final List<IREdge> sameTagOutEdges : groupOutEdgesByAdditionalOutputTag(outEdges).values()) {
```

Comment:
It might be better to keep this line and the private method.

Reply:
I'll refactor the private method then :)